front/lib/resources/workspace_resource.ts

```diff
@@ -60,11 +60,94 @@ export class WorkspaceResource extends BaseResource<WorkspaceModel> {
     return workspaces.map((workspace) => new this(this.model, workspace.get()));
   }
 
+  static async fetchByWorkOSOrganizationId(
+    workOSOrganizationId: string
+  ): Promise<WorkspaceResource | null> {
+    const workspace = await this.model.findOne({
+      where: { workOSOrganizationId },
+    });
+    return workspace ? new this(this.model, workspace.get()) : null;
+  }
+
   static async listAll(): Promise<WorkspaceResource[]> {
     const workspaces = await this.model.findAll();
     return workspaces.map((workspace) => new this(this.model, workspace.get()));
   }
 
+  async updateSegmentation(segmentation: WorkspaceSegmentationType) {
+    return this.update({ segmentation });
+  }
+
+  static async updateName(
```

Comment:
Normally the pattern is that fetch/list methods are static, while update methods take an instance. Here, I end up with several static update methods because the caller doesn't have a `WorkspaceResource` but has a `LightWorkspaceType`. And we really just need the id to do the update, so it seems artificial to build up a `WorkspaceResource` from the `LightWorkspaceType` just for the sake of having an instance method.

Any other approach here to avoid static update methods?

Reply:
Same for deletion. There are places that have a `WorkspaceType`, and call:

```typescript
  await WorkspaceModel.destroy({
    where: {
      id: workspace.id,
    },
  });
```

Right now `WorkspaceResource` has an instance `delete` method which won't work, as this might call for a static `delete` that takes id.

Or should the goal be to try to somehow merge `WorkspaceType` (and `LightWorkspaceType`) with `WorkspaceResource`? I'm thinking not, as `WorkspaceResource` is very close to the DB, while `WorkspaceType` has things like roles/groups which are separate. But having all these similar types feels a bit messy.

Reply:
Ok, but in the meantime, I don't see a clean way of avoiding class methods.